addons/im_livechat/controllers/thread.py

```diff
@@ -0,0 +1,12 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route
+from odoo.addons.mail.controllers import thread
+
+
+class ThreadController(thread.ThreadController):
+    @route()
+    def mail_message_post(self, thread_model, thread_id, post_data, context=None, **kwargs):
+        if selected_answer_id := kwargs.pop("selected_answer_id", None):
+            request.update_context(selected_answer_id=selected_answer_id)
+        return super().mail_message_post(thread_model, thread_id, post_data, context, **kwargs)
```

Comment:
As I said in a previous comment, ok for FIX, but probably at some point feels more natural as a RPC for selecting answer rather than `message_post`.

This would prevent `update_context` and jumping in different parts of the flow for finding where the context data is inserted and retrieved: everything would respect more Locality of Concern in the same route.